crates/relayer/src/chain/cosmos.rs

```diff
@@ -144,12 +145,12 @@ pub struct CosmosSdkChain {
     grpc_addr: Uri,
     light_client: TmLightClient,
     rt: Arc<TokioRuntime>,
-    keybase: KeyRing<Secp256k1KeyPair>,
+    keybase: RwLock<KeyRing<Secp256k1KeyPair>>,
```

Comment:
I wonder if we are not better off pushing the lock inside the `KeyRing` instead of holding it here, what do you think?